### PR TITLE
Add Acceptance tests and documentation for kustomization resource

### PR DESCRIPTION
**Pull request by Shubbhang351**
1. **What this PR does / why we need it**:

Add Acceptance tests and documentation for kustomization resource

2. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

3. **Additional information**

4. **Special notes for your reviewer**

     Output for acceptance test
     
     `
   === RUN   TestAcceptanceForKustomizationResource
    resource_kustomization_test.go:73: start kustomization resource acceptance tests!
TMC resources applied to the cluster successfully
    resource_kustomization_test.go:102: kustomization resource acceptance test complete
--- PASS: TestAcceptanceForKustomizationResource (196.76s)
PASS
ok      github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kustomization     198.405s

    `

### CODE REVIEW

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]
Same comments as in the git repository MR https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/168#pullrequestreview-1471213136

**Comment by ishangupta-ds** [comment]
@Shubbhang351 can you please add a guide on how to use GitOps from the Terraform provider as part of adding the docs for kustomization since it is the last resource.

**Comment by narahari92** [comment]
LGTM

**Comment by Shubbhang351** [comment]
> @Shubbhang351 can you please add a guide on how to use GitOps from the Terraform provider as part of adding the docs for kustomization since it is the last resource.

added PTAL in docs

File: internal/resources/kustomization/resource_kustomization_test.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+// nolint: dupl
+func TestAcceptanceForKustomizationResource(t *testing.T) {
+	testConfig := testGetDefaultAcceptanceConfig(t)
+
+	t.Log("start kustomization resource acceptance tests!")
+
+	// Test case for kustomization resource for GO_GIT git implementation type.
+	resource.Test(t, resource.TestCase{
```
Comment (Vasundharashukla) [inline internal/resources/kustomization/resource_kustomization_test.go:76]: Take reference of network policy PR and add update scenario in the tests. 
cc: @ramya-bangera

File: internal/resources/kustomization/resource_kustomization_test.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+func (testConfig *testAcceptanceConfig) getTestKustomizationResourceBasicConfigValue(scope commonscope.Scope) string {
+	helperBlock, scopeBlock := testConfig.ScopeHelperResources.GetTestResourceHelperAndScope(scope, kustomizationscope.ScopesAllowed[:])
+
+	return fmt.Sprintf(`
+	%s
+	
+	resource "%s" "%s" {
+	 name = "%s"
+
+	 namespace_name = "tanzu-continuousdelivery-resources"
```
Comment (Vasundharashukla) [inline internal/resources/kustomization/resource_kustomization_test.go:110]: Should this be hardcoded?

File: templates/resources/kustomization.md.tmpl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+The `tanzu-mission-control_kustomization` resource allows you to add, update, and delete Kustomization to a particular scope through Tanzu Mission Control.
+
+To create a kustomization, you must be associated with the cluster.admin or clustergroup.admin role
+
+In Creation of kustomization we must required to create Git Repository first, which we need to referenced in spec of kustomization, Git Repository can be created by using "tanzu-mission-control_git_repository" resource from terraform provider itself.
+
+[kustomization]: https://docs.vmware.com/en/VMware-Tanzu-Mission-Control/services/tanzumc-using/GUID-99916A6D-5DAF-4A26-88C7-28662F847F2F.html
+
+## kustomization Scope
```
Comment (Vasundharashukla) [inline templates/resources/kustomization.md.tmpl:17]: Is this the right heading, please check. Also, `Kustomization`

**Review by Vasundharashukla — CHANGES_REQUESTED** [review]

**Comment by Vasundharashukla** [comment]
@Shubbhang351 are the resource templates present for this resource already? If not please add the same. 
cc: @ishangupta-ds

**Comment by Shubbhang351** [comment]
> @Shubbhang351 are the resource templates present for this resource already? If not please add the same. cc: @ishangupta-ds

[CG-kustomization](https://github.com/vmware/terraform-provider-tanzu-mission-control/blob/feat/fluxcd-gitops/resource_templates/cluster_group_scoped_kustomization.tf)
[C-kustomization](https://github.com/vmware/terraform-provider-tanzu-mission-control/blob/feat/fluxcd-gitops/resource_templates/cluster_scoped_kustomization.tf)

File: docs/resources/kustomization.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    source {
+        name = "testGitRepositoryName" # Required
+        namespace = "testGitRepositoryNamespace" # Required
+    }
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply docs/resources/kustomization.md:100]: yes

File: templates/resources/kustomization.md.tmpl (hunk 3 above)
Comment (Shubbhang351) [reply templates/resources/kustomization.md.tmpl]: Updated

File: internal/resources/kustomization/resource_kustomization_test.go (hunk 2 above)
Comment (Shubbhang351) [reply internal/resources/kustomization/resource_kustomization_test.go]: updated PTAL

File: internal/resources/kustomization/resource_kustomization_test.go (hunk 1 above)
Comment (Shubbhang351) [reply internal/resources/kustomization/resource_kustomization_test.go:76]: added PTAL

File: internal/resources/kustomization/resource_kustomization_test.go [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+		KustomizationResourceName: fmt.Sprintf("%s.%s", kustomizationResource, kustomizationResourceVar),
+		KustomizationName:         acctest.RandomWithPrefix(kustomizationNamePrefix),
+		ScopeHelperResources:      commonscope.NewScopeHelperResources(),
+		GitRepositoryResource:     gitRepositoryResource,
+		GitRepositoryResourceVar:  gitRepositoryResourceVar,
+		GitRepositoryName:         acctest.RandomWithPrefix(gitRepositoryNamePrefix),
+	}
+}
+
+// nolint: dupl
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/kustomization/resource_kustomization_test.go]: removed PTAL

File: internal/resources/kustomization/resource_kustomization_test.go [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+				Check:  testConfig.checkKustomizationResourceAttributes(commonscope.ClusterScope),
+			},
+			{
+				Config: testConfig.getTestKustomizationResourceBasicConfigValue(commonscope.ClusterScope, WithPruneEnabled(true), WithInterval("10m")),
+				Check:  testConfig.checkKustomizationResourceAttributes(commonscope.ClusterScope),
+			},
+			{
+				Config: testConfig.getTestKustomizationResourceBasicConfigValue(commonscope.ClusterGroupScope),
+				Check:  testConfig.checkKustomizationResourceAttributes(commonscope.ClusterGroupScope),
+			},
```
Comment (Vasundharashukla) [inline internal/resources/kustomization/resource_kustomization_test.go:97]: Add similar update for cluster group scope as well.

**Review by Vasundharashukla — CHANGES_REQUESTED** [review]

File: internal/resources/kustomization/resource_kustomization_test.go (hunk 6 above)
Comment (Shubbhang351) [reply internal/resources/kustomization/resource_kustomization_test.go:97]: added  and tested PTAL

`
=== RUN   TestAcceptanceForKustomizationResource
    resource_kustomization_test.go:73: start kustomization resource acceptance tests!
TMC resources applied to the cluster successfully
    resource_kustomization_test.go:106: kustomization resource acceptance test complete
--- PASS: TestAcceptanceForKustomizationResource (224.57s)
PASS
ok      github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kustomization     226.067s
`

**Review by Vasundharashukla — APPROVED** [review]

File: docs/resources/kustomization.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+- `id` (String) The ID of this resource.
+- `status` (Map of String) Status for the Kustomization.
+
+<a id="nestedblock--scope"></a>
+### Nested Schema for `scope`
+
+Optional:
+
+- `cluster` (Block List, Max: 1) The schema for cluster iam policy full name (see [below for nested schema](#nestedblock--scope--cluster))
+- `cluster_group` (Block List, Max: 1) The schema for cluster group iam policy full name (see [below for nested schema](#nestedblock--scope--cluster_group))
```
Comment (ishangupta-ds) [inline docs/resources/kustomization.md:124]: Why is kustomization documentation having IAM policy?

**Review by ishangupta-ds — CHANGES_REQUESTED** [review]

File: docs/resources/kustomization.md (hunk 7 above)
Comment (Shubbhang351) [reply docs/resources/kustomization.md]: removed, updated the docs

**Pull request merged**